components/org.wso2.carbon.identity.scim.v2.common/src/main/java/org/wso2/carbon/identity/scim.v2/common/DAO/GroupDAO.java

```diff
@@ -0,0 +1,325 @@
+
+        try {
+            prepStmt = connection.prepareStatement(SQLQueries.CHECK_EXISTING_GROUP_SQL);
+            prepStmt.setInt(1, tenantId);
+            prepStmt.setString(2, SCIMCommonUtils.getGroupNameWithDomain(groupName));
+
+            rSet = prepStmt.executeQuery();
+            if (rSet.next()) {
+                isExistingGroup = true;
+            }
+            connection.commit();
+        } catch (SQLException e) {
+            throw new IdentitySCIMException("Error when reading the group information from the persistence store.", e);
+        } finally {
+            IdentityDatabaseUtil.closeAllConnections(connection, rSet, prepStmt);
+        }
+        return isExistingGroup;
+    }
+
+    private boolean isExistingAttribute(String attributeName, String groupName, int tenantId)
+            throws IdentitySCIMException {
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement prepStmt = null;
+        ResultSet rSet = null;
+
+        boolean isExistingAttribute = false;
+
+        try {
+            prepStmt = connection.prepareStatement(SQLQueries.CHECK_EXISTING_ATTRIBUTE_SQL);
+            prepStmt.setInt(1, tenantId);
+            prepStmt.setString(2, SCIMCommonUtils.getGroupNameWithDomain(groupName));
+            prepStmt.setString(3, attributeName);
+
+            rSet = prepStmt.executeQuery();
+            if (rSet.next()) {
+                isExistingAttribute = true;
+            }
+            connection.commit();
+        } catch (SQLException e) {
+            throw new IdentitySCIMException("Error when reading the group attribute information from " +
+                    "the persistence store.", e);
+        } finally {
+            IdentityDatabaseUtil.closeAllConnections(connection, rSet, prepStmt);
+        }
+        return isExistingAttribute;
+    }
+
+    public void addSCIMGroupAttributes(int tenantId, String roleName, Map<String, String> attributes)
+            throws IdentitySCIMException {
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement prepStmt = null;
+
+        if (!isExistingGroup(SCIMCommonUtils.getGroupNameWithDomain(roleName), tenantId)) {
+            try {
+                prepStmt = connection.prepareStatement(SQLQueries.ADD_ATTRIBUTES_SQL);
+                prepStmt.setInt(1, tenantId);
+                prepStmt.setString(2, roleName);
+                int count = 0;
+                for (Map.Entry<String, String> entry : attributes.entrySet()) {
+                    if (!isExistingAttribute(entry.getKey(),
+                            SCIMCommonUtils.getGroupNameWithDomain(roleName), tenantId)) {
+                        prepStmt.setString(3, entry.getKey());
+                        prepStmt.setString(4, entry.getValue());
+
+                        prepStmt.addBatch();
+                        count++;
+
+                        if (count == attributes.size()) {
```

Comment:
I think we can get rid of this check and do the executing of the batch outside the for-loop. Similar implementation can be found [here](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java#L1072).